File: sdk/go/common/util/cmdutil/testdata/frozen.go
//go:build ignore
// +build ignore

package main

import (
	"fmt"
	"log"
	"os"
	"os/signal"
	"time"
)

func main() {
	log.SetFlags(0)

	sigch := make(chan os.Signal, 1)
	signal.Notify(sigch, os.Interrupt)
	// os.Interrupt handles SIGINT and CTRL_BREAK_EVENT.

	fmt.Println("ready")
	select {
	case <-sigch:
		time.Sleep(3 * time.Second)
		log.Fatal("error: was not forced to exit")

	case <-time.After(3 * time.Second):
		log.Fatal("error: did not receive signal")
	}
}

File: sdk/go/common/util/cmdutil/testdata/frozen.js
function handleSignal() {
	setTimeout(function() {
		console.errro("error: was not forced to exit");
		process.exit(2);
	}, 3000);
}

process.on('SIGINT', handleSignal);
process.on('SIGBREAK', handleSignal); // ctrl-break on windows
console.log('ready');

setTimeout(function() {
	console.error('error: did not receive signal');
	process.exit(1);
}, 3000);
